Doc/c-api/exceptions.rst

```diff
@@ -505,29 +505,72 @@ Signal Handling
       single: SIGINT
       single: KeyboardInterrupt (built-in exception)
 
-   This function interacts with Python's signal handling.  It checks whether a
-   signal has been sent to the processes and if so, invokes the corresponding
-   signal handler.  If the :mod:`signal` module is supported, this can invoke a
-   signal handler written in Python.  In all cases, the default effect for
-   :const:`SIGINT` is to raise the  :exc:`KeyboardInterrupt` exception.  If an
-   exception is raised the error indicator is set and the function returns ``-1``;
-   otherwise the function returns ``0``.  The error indicator may or may not be
-   cleared if it was previously set.
+   This function interacts with Python's signal handling.
+
+   If the function is called from the main thread and under the main Python
+   interpreter, it checks whether a signal has been sent to the processes
+   and if so, invokes the corresponding signal handler.  If the :mod:`signal`
+   module is supported, this can invoke a signal handler written in Python.
+
+   If an exception is raised by a Python signal handler, the error indicator
+   is set and the function returns ``-1``; otherwise the function returns
+   ``0``.  The error indicator may or may not be cleared if it was previously
+   set.
```

Comment:
I don't get what is an "error indicator". I suggest to remove it. Same remark for PyErr_SetInterruptEx() documentation.

> If an exception is raised by a Python signal handler (...) the function returns -1

It may be worth it to mention that in this case, following Python signal handlers are not called. In this case, PyErr_CheckSignals() should be called again to call remaining Python signal handlers.

Reply:
"error indicator" is used everywhere in the C API docs :-)